endpoints-control/src/main/java/com/google/api/control/ConfigFilter.java

```diff
@@ -171,8 +176,37 @@ public static ReportingRule getReportRule(ServletRequest req) {
     return (MethodRegistry.Info) httpRequest.getAttribute(METHOD_INFO_ATTRIBUTE);
   }
 
+  /**
+   * Get the "real" HTTP method for the request, taking into account the X-HTTP-Method-Override
+   * header.
+   *
+   * @param req a {@code ServletRequest}
+   * @return the HTTP method for the request
+   */
+  public static String getRealHttpMethod(ServletRequest req) {
+    HttpServletRequest httpRequest = (HttpServletRequest) req;
+    String method = (String) httpRequest.getAttribute(HTTP_METHOD_ATTRIBUTE);
+    if (method != null) {
+      return method;
+    }
+    return httpRequest.getMethod();
+  }
+
   @Override
   public void destroy() {
     // unused
   }
+
+  private static String getRequestMethod(HttpServletRequest request) {
+    Enumeration headerNames = request.getHeaderNames();
+    String methodOverride = null;
+    while (headerNames.hasMoreElements()) {
+      String headerName = (String) headerNames.nextElement();
+      if (headerName.toLowerCase().equals("x-http-method-override")) {
+        methodOverride = request.getHeader(headerName);
+        break;
+      }
+    }
+    return methodOverride != null ? methodOverride.toUpperCase() : request.getMethod();
```

Comment:
I think this should just return `methodOverride`, since `getRealHttpMethod` accounts for the possibility this is null.

That is, you're checking the specified http method in two places; here and in `getRealHttpMethod`. I think you should check in only one place; have this method return the method override, if any, and have `getRealHttpMethod` use the appropriate value. (You would probably also want to rename this method to something like `getRequestMethodOverride`

Reply:
Good call. Done.